lua/nvim_lsp/health.lua

```diff
@@ -2,19 +2,25 @@ local M = {}
 function M.check_health()
   local configs = require 'nvim_lsp/configs'
 
-  for _, top_level_config in pairs(configs) do
+  for _, default_config in pairs(configs) do
     -- the folder needs to exist
-    local new_config = top_level_config.make_config(".")
+    local config = default_config.make_config(".")
 
-    local cmd, _ = vim.lsp._cmd_parts(new_config.cmd)
-    if not (vim.fn.executable(cmd) == 1) then
+    local status, cmd = pcall(vim.lsp._cmd_parts, config.cmd)
+    if not status then
       vim.fn['health#report_error'](
-        string.format("%s: The given command %q is not executable.", new_config.name, cmd)
+        string.format("%s: config.cmd, %s, is expected 'table', but got %s", config.name, vim.inspect(config.cmd), type(config.cmd))
```

Comment:
this may hide a real error, pcall should return the error message (`cmd`) if an error happens so use that in the message instead of assuming the error.